Clarify importance of options when protecting from recipe changes (#9548)
conan-io/conan-center-index: docs/faqs.md

need to modify your project if the recipe changes the binaries, flags,... it provides).

To isolate from this changes there are different strategies you can follow:

> The minimum solution, involves small changes to you Conan client configuration by

Reviewer: ```suggestion
The minimum solution involves small changes to your Conan client configuration by
```

>  * **Pin the version of every reference you consume in your project** using recipe revisions
>    and lockfiles.

Reviewer 2: ```suggestion
 * **Pin the version of every reference you consume in your project** using recipe [revisions](https://docs.conan.io/en/latest/versioning/revisions.html)
   and [lockfiles](https://docs.conan.io/en/latest/versioning/lockfiles/introduction.html).
```

>  * **Pin the version of every reference you consume in your project** using recipe [revisions](https://docs.conan.io/en/latest/versioning/revisions.html)

Reviewer 3: ```suggestion
 * **Pin the version of every reference you consume in your project** using [recipe revision (RREV)](https://docs.conan.io/en/latest/versioning/revisions.html)
```

> For larger projects and teams it is recommended to add some infratructure to ensure stability by

Reviewer: ```suggestion
For larger projects and teams it is recommended to add some infrastructure to ensure stability by
```

> 

Reviewer 3: It's quite vague for many people I think. A link to documentation with a concrete example would help.

Author: Theres a link below to a page with more documentation just outside of the diff.

If you didn't see that, I can move it up to make it more obvious

Reviewer 4: links should be inline, the word `revisions` should point to the revisions doc, and the word `lockfile` should point to the lockfiles doc. for newbies, these words probably say nothing itself.

>  * **Pin the version of every reference you consume in your project** using recipe [revisions](https://docs.conan.io/en/latest/versioning/revisions.html)
>    and [lockfiles](https://docs.conan.io/en/latest/versioning/lockfiles/introduction.html).

Reviewer 3: ```suggestion
* **Pin the version of every reference you consume in your project** using either:
  * r[recipe revision (RREV)](https://docs.conan.io/en/latest/versioning/revisions.html): `foo/1.0@#RREV` instead of `foo/1.0`
  * [lockfiles](https://docs.conan.io/en/latest/versioning/lockfiles/introduction.html)
```

More explicit, new comers can easily get confused by RREV and PREV.
